pylatex/table.py

```diff
@@ -31,13 +35,15 @@ def _get_table_width(table_spec):
         The width of a table which uses the specification supplied.
     """
 
-    column_letters = ['l', 'c', 'r', 'p', 'm', 'b']
-
     # Remove things like {\bfseries}
     cleaner_spec = re.sub(r'{[^}]*}', '', table_spec)
+
+    # Remove X[] in tabu environments so they dont interfere with column count
+    for l in COLUMN_LETTERS:
+        cleaner_spec = re.sub(r'X\[[^X]*' + l + r'\]', l, cleaner_spec)
```

Comment:
could you change this call to this?

```python
cleaner_spec = re.sub(r'X\[(.*?)\]', r'\1', cleaner_spec)
```

It should be a little less error prone.